src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/8.diff.sql

```diff
@@ -61,6 +61,105 @@ WITH
     ONLINE = ON
 )
 
+CREATE TYPE dbo.DateTimeSearchParamTableType_2 AS TABLE
+(
+    SearchParamId smallint NOT NULL,
+    StartDateTime datetimeoffset(7) NOT NULL,
+    EndDateTime datetimeoffset(7) NOT NULL,
+    IsLongerThanADay bit NOT NULL,
+    IsMin bit NOT NULL,
+    IsMax bit NOT NULL
+);
+
+ALTER TABLE dbo.DateTimeSearchParam
+ADD IsMin bit NOT NULL,
+    IsMax bit NOT NULL,
+    CONSTRAINT date_IsMin_Constraint DEFAULT 0 FOR IsMin,
```

Comment:
> CONSTRAINT date_IsMin_Constraint DEFAULT 0 FOR IsMin, [](start = 3, length = 54)

I've tried to find what Default does for column, but I don't see that option in docs: [https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-column-constraint-transact-sql?view=sql-server-ver15](https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-table-column-constraint-transact-sql?view=sql-server-ver15)

Can you comment what it does and in case this to setup default value for added column why not: `ADD IsMin bit NOT NULL DEFAULT(0)`

Reply:
I did attempt that approach as well and there is no difference in functionality. In both scenarios, the end result is that we create a `DefaultConstraint` object on the table. The advantage of specifically adding a `CONSTRAINT` is that we can set the name. This makes it cleaner to test (we can specifically exclude the named constraint in our schema difference test - `GivenTwoSchemaInitializationMethods_WhenCreatingTwoDatabases_BothSchemasShouldBeEquivalent()`). If we use `DEFAULT(0)`, the constraint doesn't have a name.